src/Microsoft.Health.Dicom.Operations/Registration/ServiceCollectionExtensions.cs

```diff
@@ -11,49 +11,63 @@
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.DependencyInjection.Extensions;
 using Microsoft.Health.Blob.Configs;
-using Microsoft.Health.Dicom.Functions.Configuration;
-using Microsoft.Health.Dicom.Functions.Durable;
-using Microsoft.Health.Dicom.Functions.Registration;
+using Microsoft.Health.Dicom.Core.Configs;
+using Microsoft.Health.Dicom.Core.Modules;
+using Microsoft.Health.Dicom.Core.Registration;
+using Microsoft.Health.Dicom.Operations.Configuration;
+using Microsoft.Health.Dicom.Operations.Durable;
+using Microsoft.Health.Dicom.Operations.Indexing;
+using Microsoft.Health.Dicom.Operations.Management;
+using Microsoft.Health.Dicom.Operations.Registration;
+using Microsoft.Health.Extensions.DependencyInjection;
 using Microsoft.Health.SqlServer.Configs;
 using Microsoft.IO;
 
 namespace Microsoft.Extensions.DependencyInjection
 {
-    internal static class ServiceCollectionExtensions
+    public static class ServiceCollectionExtensions
     {
-        public static IServiceCollection AddFunctionsOptions<T>(
+        public static IDicomFunctionsBuilder ConfigureFunctions(
             this IServiceCollection services,
-            IConfiguration configuration,
-            string sectionName,
-            bool bindNonPublicProperties = false)
-            where T : class
+            IConfiguration configuration)
         {
             EnsureArg.IsNotNull(services, nameof(services));
             EnsureArg.IsNotNull(configuration, nameof(configuration));
-            EnsureArg.IsNotEmptyOrWhiteSpace(sectionName, nameof(sectionName));
 
-            services
-                .AddOptions<T>()
-                .Bind(
-                    configuration
-                        .GetSection(DicomFunctionsConfiguration.SectionName)
-                        .GetSection(sectionName),
-                    x => x.BindNonPublicProperties = bindNonPublicProperties)
-                .ValidateDataAnnotations();
+            services.RegisterModule<ServiceModule>(new FeatureConfiguration { EnableExtendedQueryTags = true });
 
-            return services;
+            return new DicomFunctionsBuilder(services
```

Comment:
This file has seen a number of refactors to allow for reuse between here and a PAAS equivalent.